test/001_highsnr_stdio_audio/test_rx.py

```diff
@@ -0,0 +1,180 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Wed Dec 23 07:04:24 2020
+
+@author: DJ2LS
+"""
+
+import ctypes
+from ctypes import *
+import pathlib
+import pyaudio
+import sys
+import logging
+import time
+import threading
+import sys
+import argparse
+import numpy as np
+sys.path.insert(0,'..')
+import codec2
+
+#--------------------------------------------GET PARAMETER INPUTS  
+parser = argparse.ArgumentParser(description='Simons TEST TNC')
+parser.add_argument('--bursts', dest="N_BURSTS", default=1, type=int)
+parser.add_argument('--framesperburst', dest="N_FRAMES_PER_BURST", default=1, type=int)
+parser.add_argument('--mode', dest="FREEDV_MODE", type=str, choices=['datac0', 'datac1', 'datac3'])
+parser.add_argument('--audiodev', dest="AUDIO_INPUT_DEVICE", default=-1, type=int,
+                    help="audio device number to use, use -2 to automatically select a loopback device")  
+parser.add_argument('--debug', dest="DEBUGGING_MODE", action="store_true")  
+parser.add_argument('--timeout', dest="TIMEOUT", default=10, type=int, help="Timeout (seconds) before test ends")  
+parser.add_argument('--list', dest="LIST", action="store_true", help="list audio devices by number and exit")  
+
+args = parser.parse_args()
+
+if args.LIST:
+    p = pyaudio.PyAudio()
+    for dev in range(0,p.get_device_count()):
+        print("audiodev: ", dev, p.get_device_info_by_index(dev)["name"])
+    quit()
+   
+N_BURSTS = args.N_BURSTS
+N_FRAMES_PER_BURST = args.N_FRAMES_PER_BURST
+AUDIO_INPUT_DEVICE = args.AUDIO_INPUT_DEVICE
+MODE = codec2.FREEDV_MODE[args.FREEDV_MODE].value
+DEBUGGING_MODE = args.DEBUGGING_MODE
+TIMEOUT = args.TIMEOUT
+
+# AUDIO PARAMETERS
+AUDIO_FRAMES_PER_BUFFER = 2400*2  # <- consider increasing if you get nread_exceptions > 0
```

Comment:
I needed to fiddle with this to stop audio break up :point_up: Note it also needs to be a multiple of 48/8=6 for the resampler to work.